interpreter/interpretable.go

```diff
@@ -214,41 +213,46 @@ func (or *evalOr) ID() int64 {
 
 // Eval implements the Interpretable interface method.
 func (or *evalOr) Eval(ctx Activation) ref.Val {
-	// short-circuit lhs.
-	lVal := or.lhs.Eval(ctx)
-	lBool, lok := lVal.(types.Bool)
-	if lok && lBool == types.True {
-		return types.True
-	}
-	// short-circuit on rhs.
-	rVal := or.rhs.Eval(ctx)
-	rBool, rok := rVal.(types.Bool)
-	if rok && rBool == types.True {
-		return types.True
-	}
-	// return if both sides are bool false.
-	if lok && rok {
-		return types.False
-	}
-	// TODO: return both values as a set if both are unknown or error.
-	// prefer left unknown to right unknown.
-	if types.IsUnknown(lVal) {
-		return lVal
+	var err ref.Val = nil
+	var unk types.Unknown = nil
+	for _, term := range or.terms {
+		val := term.Eval(ctx)
+		boolVal, ok := val.(types.Bool)
+		// short-circuit on true.
+		if ok && boolVal == types.True {
+			return types.True
+		}
+		if !ok {
+			if types.IsUnknown(val) {
+				if unk == nil {
+					unk = types.Unknown{}
+				}
+				unk = append(unk, val.(types.Unknown)...)
+				continue
+			}
+			if types.IsError(val) {
+				if unk == nil && err == nil {
+					err = val
+				}
+				continue
+			}
+			if unk == nil {
+				err = types.MaybeNoSuchOverloadErr(val)
+			}
```

Comment:
We might be getting to the point where we should extract the unknown and error handling into a function.

Reply:
I've added a helper in types/unknown.go and simplified the logic here. PTAL